File: c/lcdconvert.c
#include "stdlib.h"
#include "stdio.h"
#include "ctype.h"
#include "string.h"

const char* alphabet[3] = {
    " _     _  _     _  _  _  _  _ ",
    "| |  | _| _||_||_ |_   ||_||_|",
    "|_|  ||_  _|  | _||_|  ||_| _|"
};

/**
 * Display the given string in ASCII LCD format
*/
void displayLcd(char *s) {
  // Display the lines one by one
  for (int line = 0; line < 3; line++) {
    for (int c = 0; c < strlen(s); c++) {
      for (int col = 0; col < 3; col++) {
        // substract 48 to convert ASCII to numerical
        printf("%c", (char) alphabet[line][(s[c] - 48) * 3 + col]);
      }
    }
    printf("\n");
  }
}

/**
 * Display program usage
*/
void showUsage(char *name) {
  printf("Usage: %s [integer]\n", name);
}

/**
 * Check user arguments and call displayLcd with first argument
*/
int main(int argc, char **argv) {
  // Verify argument count
  if (argc != 2) {
    showUsage(argv[0]);
    return EXIT_FAILURE;
  }

  // Verify that first argument is a number
  for (int i = 0; i < strlen(argv[1]); i++) {
    if (!isdigit(argv[1][i])) {
      showUsage(argv[0]);
      return EXIT_FAILURE;
    }
  }

  displayLcd(argv[1]);
  return EXIT_SUCCESS;
}

File: python/lcdconvert.py
import sys

# ASCII LCD alphabet
alphabet = [
    " _     _  _     _  _  _  _  _ ",
    "| |  | _| _||_||_ |_   ||_||_|",
    "|_|  ||_  _|  | _||_|  ||_| _|"
]

#
# Appends the next n number to the ASCII LCD buffer
#
def append_lcd_num(n, buffer):
  for i in range(3):
    for j in range(3):
      buffer[i].append(alphabet[i][j + 3 * n])

#
# Display the given n number in ASCII LCD format
#
def display_lcd(n):
  stringified_n = str(n)
  final_output = [[], [], []]

  # Create the final output string by appending numbers
  for i in range(len(stringified_n)):
    append_lcd_num(int(stringified_n[i]), final_output)

  # Display the final output string
  for i in range(3):
    print("".join(final_output[i]))

#
# Display usage to user
#
def show_usage():
   print(f"Usage : python3 {sys.argv[0]} [integer]")

#
# Main function, calling display_lcd with the first argument number
#
def main():
  if len(sys.argv) != 2:
    show_usage()
    return

  try:
    n = int(sys.argv[1])
  except:
    print("Please enter a number as the only argument.")
    return
  display_lcd(n)

#
# Only execute this if we are in a CLI context
#
if __name__ == "__main__":
    main()
